src/components/ProfileList/ProfileList.tsx

```diff
@@ -82,42 +55,42 @@ const ProfileListItem: React.FC<{ data: ProfileListItems }> = ({ data }) => {
         </div>
         <div className={styles.data}>
           <p className={styles.name}>{data.name}</p>
-          <p className={seniorityClass}>{data.seniority}</p>
-          <p className={styles.location}>{data.location}</p>
+          <p className={getStackClasses}>
+            {data.seniority} {data.stack} Developer
+          </p>
+          <p className={styles.location}>
+            {data.country}, {data.city} / {data.remote}
+          </p>
         </div>
       </div>
-      <div className={styles.technology}>
-        <TechnologyChip
-          name={data.technology}
-          theme={theme}
-          seniority={data.seniority}
-        />
-        <TechnologyChip
-          name={data.technology2}
-          theme={theme}
-          seniority={data.seniority}
-        />
-        <TechnologyChip
-          name={data.technology3}
-          theme={theme}
-          seniority={data.seniority}
-        />
-        <TechnologyChip
-          name={data.others}
-          theme={theme}
-          seniority={data.seniority}
-        />
-      </div>
+      <div className={getTechnologyClasses}>{renderTechnologies()}</div>
     </div>
   )
 }
 
 const ProfileList: React.FC = () => {
+  const {
+    stackFilter,
+    technologyFilter,
+    seniorityFilter,
+    availabilityFilter,
+    locationFilter,
+  } = useFilters()
+  const filteredProfileData = ProfileData.filter((profile) => {
+    return (
+      (!stackFilter || profile.stack === stackFilter) &&
+      (!seniorityFilter || profile.seniority === seniorityFilter) &&
+      (!locationFilter || profile.country === locationFilter) &&
```

Comment:
**nitpick:** We should split this, as it's non maintainable, and quite messy. Let's chain these filters, so it's cleanly understandable which filter does what. Let's create filter chain as following:

```js
const filteredProfileData = profileData.filter(filterByStack(stackFilter)).filter(filterBySeniority(seniorityFilter))
...
```